client/state/plugins/wporg/test/actions.js

```diff
@@ -14,6 +14,8 @@ jest.mock( 'lib/impure-lodash', () => ( {
 	debounce: cb => cb,
 } ) );
 
+jest.useFakeTimers();
```

Comment:
using fake timers fixed a leak warning

Reply:
The only reason why we need the timers here is that the `fetchPluginData` calls `setTimeout( dispatch( ... ), 1 )`. That's a Fluxism that's not necessary in Redux. See https://github.com/Automattic/wp-calypso/pull/2018/files#r50295012 where it was introduced during a Flux to Redux migration by @johnHackworth 

We should fix the code instead of hacking the tests.

Reply:
You should also consider using fake timers for all tests by default as it allows to detect such issues in the first place :)

Reply:
The fake timers implementation depends on `@jest-environment`: it's different for JSDOM and for Node.

Also, for some tests the Jest implementation is insufficient and we need to use the Sinon one. The difference is whether `Date.now()` is faked or not. Jest doesn't fake the current time. Lodash `throttle` and `debounce` expect `Date.now()` to advance by 1 second after `setTimeout( _, 1000 )` and therefore need the fake timer in order to work at all. See also #28749.

Both issues may or may not be an obstacle for the fake-by-default approach. Just be prepared to encounter them 🙂

Reply:
Regarding non deterministic lodash functions, see https://github.com/Automattic/wp-calypso/blob/master/client/lib/impure-lodash/README.md

Reply:
I don't see how `impure-lodash` solves the problem with fake timers. 🙂

Reply:
I'm fixing the wporg fetch the right way in #28915

Reply:
Deferred the work there, removed from this PR